### DISCUSSION TITLE
Do I need Django?

### DISCUSSION THREAD

**Discussion by RickkeeC** [infiniteremote/installer#11] [Q&A]
First time I installed, I was prompted for web password.  Poked around a few hours and couldn't get it to work.  Tried uninstalling from rustdesk notes, reinstall and now the mess. 
Last metadata expiration check: 3:13:59 ago on Mon 16 Sep 2024 07:08:57 PM EDT.
Dependencies resolved.
Nothing to do.
Complete!
Last metadata expiration check: 3:14:09 ago on Mon 16 Sep 2024 07:08:57 PM EDT.
Package curl-7.76.1-31.el9.x86_64 is already installed.
Package wget-1.21.1-8.el9.x86_64 is already installed.
Package unzip-6.0-57.el9.x86_64 is already installed.
Package tar-2:1.34-7.el9.x86_64 is already installed.
Package git-2.43.5-1.el9.x86_64 is already installed.
No match for argument: python3.9-venv
Package bind-utils-32:9.16.23-24.el9.x86_64 is already installed.
Error: Unable to find a match: python3.9-venv
sudo: ufw: command not found
sudo: ufw: command not found
sudo: ufw: command not found
sudo: ufw: command not found
Installing RustDesk Server
--2024-09-16 22:23:10--  https://github.com/rustdesk/rustdesk-server/releases/download/1.1.11-1/rustdesk-server-linux-amd64.zip
Resolving github.com (github.com)... 140.82.113.3
Connecting to github.com (github.com)|140.82.113.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/299354666/0285ef87-85f9-441e-b8b8-8efc8c4f977b?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240917%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240917T022310Z&X-Amz-Expires=300&X-Amz-Signature=324dc79093d4d9c489f011a01c964ba8bb230a8184758a1e0d9ba11666f19a48&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=299354666&response-content-disposition=attachment%3B%20filename%3Drustdesk-server-linux-amd64.zip&response-content-type=application%2Foctet-stream [following]
--2024-09-16 22:23:10--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/299354666/0285ef87-85f9-441e-b8b8-8efc8c4f977b?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240917%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240917T022310Z&X-Amz-Expires=300&X-Amz-Signature=324dc79093d4d9c489f011a01c964ba8bb230a8184758a1e0d9ba11666f19a48&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=299354666&response-content-disposition=attachment%3B%20filename%3Drustdesk-server-linux-amd64.zip&response-content-type=application%2Foctet-stream
Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.109.133, ...
Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8539403 (8.1M) [application/octet-stream]
Saving to: ‘rustdesk-server-linux-amd64.zip’

rustdesk-server-linux-amd64.zip   100%[===========================================================>]   8.14M  11.1MB/s    in 0.7s    

2024-09-16 22:23:11 (11.1 MB/s) - ‘rustdesk-server-linux-amd64.zip’ saved [8539403/8539403]

Archive:  rustdesk-server-linux-amd64.zip
  inflating: amd64/hbbr              
  inflating: amd64/hbbs              
  inflating: amd64/rustdesk-utils    
RustDesk Relay not ready yet...
Tidying up install
./install.sh: line 230: cd: /opt: No such file or directory
fatal: destination path 'rustdesk-api-server' already exists and is not an empty directory.
chown: cannot access '/opt/rustdesk-api-server/': No such file or directory
./install.sh: line 249: /opt/rustdesk-api-server/rustdesk_server_api/secret_config.py: No such file or directory
./install.sh: line 258: cd: /opt/rustdesk-api-server/api: No such file or directory
./install.sh: line 260: /opt/rustdesk-api-server/api/env/bin/activate: No such file or directory
./install.sh: line 261: cd: /opt/rustdesk-api-server/api/: No such file or directory
./install.sh: line 262: pip: command not found
./install.sh: line 263: pip: command not found
./install.sh: line 264: pip: command not found
./install.sh: line 265: cd: /opt/rustdesk-api-server/: No such file or directory
Traceback (most recent call last):
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 11, in main
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 22, in <module>
    main()
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 13, in main
    raise ImportError(
ImportError: Couldn't import Django. Are you sure it's installed and available on your PYTHONPATH environment variable? Did you forget to activate a virtual environment?
Traceback (most recent call last):
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 11, in main
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 22, in <module>
    main()
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 13, in main
    raise ImportError(
ImportError: Couldn't import Django. Are you sure it's installed and available on your PYTHONPATH environment variable? Did you forget to activate a virtual environment?
Please Set your password and username for the Web UI
Traceback (most recent call last):
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 11, in main
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 22, in <module>
    main()
  File "/var/lib/rustdesk-server/rustdesk-api-server/manage.py", line 13, in main
    raise ImportError(
ImportError: Couldn't import Django. Are you sure it's installed and available on your PYTHONPATH environment variable? Did you forget to activate a virtual environment?
./install.sh: line 270: deactivate: command not found
tee: /opt/rustdesk-api-server/api/api_config.py: No such file or directory
Installing nginx
Last metadata expiration check: 3:14:36 ago on Mon 16 Sep 2024 07:08:57 PM EDT.
Dependencies resolved.
======================================================================================================================================
 Package                             Architecture              Version                             Repository                    Size
======================================================================================================================================
Installing:
 nginx                               x86_64                    2:1.20.1-20.el9                     appstream                     36 k
Installing dependencies:
 nginx-core                          x86_64                    2:1.20.1-20.el9                     appstream                    570 k
 nginx-filesystem                    noarch                    2:1.20.1-20.el9                     appstream                    9.2 k

**Comment by gitondezhuts** [comment]
sounds like install borked, please reinstall
